addons/mrp_subcontracting/models/stock_move.py

```diff
@@ -116,14 +116,13 @@ def _auto_record_components(self, qty):
             production._set_qty_producing()
             production.with_context(cancel_backorder=False).subcontracting_record_component()
 
-    def copy(self, default=None):
-        self.ensure_one()
-        if not self.is_subcontract or 'location_id' in default:
-            return super(StockMove, self).copy(default=default)
-        if not default:
-            default = {}
-        default['location_id'] = self.picking_id.location_id.id
-        return super(StockMove, self).copy(default=default)
+    def copy_data(self, default=None):
+        vals_list = super().copy_data(default=default)
+        for move, vals in zip(self, vals_list):
+            if 'location_id' in default or not move.is_subcontract:
```

Comment:
I just realized we did not check `default` is a dict before checking  `'location_id' in default`. It would be nice to ensure it. Also, you call `super` in the beginning now, which can impact the value of `location_id` in the new records

Reply:
Ok for first part. For the second one, I override now copy_data, which means that the records are not created yet. 

Hence I'm not sure I understand the issue here.